Feature/1748 geo backgrounds

joonasnuutinen opened the pull request:
Implements #1748.

https://user-images.githubusercontent.com/18427577/218730764-20eca7a3-36c9-4566-8902-3f0ae0779cad.mov

Note:
- if Landsat is selected, choosing year 2012 or older causes an error
- if only Sentinel is selected, choosing a year 2016 or older causes an error

The error is:
```
Pattern 'red' did not match any bands.
```

In addition, Sentinel is still quite heavy and may lead to errors if the map is zoomed out far enough.

However, these errors are related to Sepal and not the actual UI components, so maybe these should be addressed in another ticket.

Commit:
Add sepal layer

Commit:
Modify Sepal body to avoid memory error

Commit:
Update Mosaic controls based on a call with Eugenia

Commit:
Move mosaic controls to a separate component

This should improve performance because the mosaic component is now only
loaded when 'Background' is clicked

Commit:
Fix unnecessary Sepal call

Commit:
Separate UI and applied mosaic options

UI options are reflected in the UI. Applied options are the ones actually
used when fetching the background.

Commit:
Make comment more describing

Commit:
Enable apply changes button if options have changed

Commit:
Apply mosaic changes

Commit:
Set Sentinel request body based on mosaic options

Commit:
Use Landsat as default because it's faster

Commit:
Refactor styling

Commit:
Style mosaic checkboxes

Commit:
Merge branch 'development' into feature/1748-geo-backgrounds

Commit:
Fix gray map

Commit:
Change component file names to uppercase

joonasnuutinen commented:
Also, Finland is still hardcoded. Let's fix this when #2205 is merged.

yaguzmang commented on src/client/pages/Geo/GeoMap/GeoMapMenuMosaic/geoMapMenuMosaic.tsx:
```diff
@@ -1,63 +1,59 @@
+
+  // Mosaic layer toggled, mosaicUrl updated or mosaicOptions changed
+  useEffect(() => {
+    // console.log('mosaic', mosaicSelected)
```
Should we remove the console.logs?

yaguzmang commented:
Tested. Works as expected, good job! 👍

mergify[bot] committed:
Merge branch 'development' into feature/1748-geo-backgrounds

Commit:
Merge branch 'development' into feature/1748-geo-backgrounds

Commit:
Fix old method name

Commit:
Set mosaic country dynamically

Commit:
Switch mosaic on the fly if country is switched

Commit:
Move mosaic state up

The issue before was that it required the Background popup to be open.
Now the popup can be open or closed and the mosaics still load.

Commit:
Remove log comment

Commit:
Merge branch 'development' into feature/1748-geo-backgrounds

joonasnuutinen commented:
Conflicts resolved and mosaic country set dynamically.


https://user-images.githubusercontent.com/18427577/219018940-31a58cc2-49d6-432c-af4f-de02ae2394b8.mov

mergify[bot] committed:
Merge branch 'development' into feature/1748-geo-backgrounds
